glfs.c

```diff
@@ -192,6 +192,12 @@ static int gluster_cache_add(gluster_server *dst, glfs_t *fs, char* cfgstring)
 
 	return 0;
 
+free_entry:
+	if (entry->volname)
+		free(entry->volname);
+	if (entry->server)
+		free(entry->server);
```

Comment:
IMO, line 199 will never be executed, as the condition will fail always
Something similar like, 
```
diff --git a/glfs.c b/glfs.c
index 1a7fff3..fe3a525 100644
--- a/glfs.c
+++ b/glfs.c
@@ -170,8 +170,11 @@ static int gluster_cache_add(gluster_server *dst, glfs_t *fs, char* cfgstring)
        entry->volname = strdup(dst->volname);
 
        entry->server = calloc(1, sizeof(gluster_hostdef));
-       if (!entry->server)
+       if (!entry->server) {
+               free(entry->volname);
+               free(entry);
                goto error;
+       }
 
        entry->server->type = dst->server->type;
```
will be good.

@tangwenji Thanks for the catch :-)